SokolTextGame/Commands.cs

```diff
@@ -222,6 +230,63 @@ public void Execute(string[] words, World world)
             Console.ResetColor();
         }
     }
+    public class Attack : ICommand
+    {
+        public void Execute(string[] words, World world)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            var locObj = world.CurrentLocation().ObjectOnLocation;
+            if (string.Join(" ", words) == "attack" || words.Length > 2) Console.WriteLine("Use <attack> <object>");
+            else if ((words[0] == "attack" && locObj.Name == words[1] && locObj.ObjectIsAlive == false))
+            {
+                Console.WriteLine($"You've already killed {locObj.Name}.");
+            }
+            else if (words[0] == "attack" && locObj.Name == words[1] && locObj.ObjectIsAlive == true)
+            {
+                string objectName = locObj.Name;
+                int hpObject = locObj.HealtObject;
+                int hpPlayer = world.player.HealtPlayer;
+                while (hpObject > 0 || hpPlayer > 0)
+                {
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    int damageObject = new Random().Next(1, locObj.DamageObject + 1);
+                    int damagePlayer = new Random().Next(1, world.player.Weapon.WeaponDamage + 1);
+                    Console.WriteLine($"You have {hpPlayer} health");
+                    Console.WriteLine($"{objectName} have {hpObject} health");
+                    Console.WriteLine($"Write \"attack\" to attack {objectName} or \"block\" to block an {objectName} attack");
+                    string inputCommand = Console.ReadLine().ToLower().Trim();
+                    if (inputCommand == "attack")
+                    {
+                        Console.ForegroundColor = ConsoleColor.Green;
+                        Console.WriteLine($"You have dealt {damagePlayer} damage to the {objectName}.");
+                        hpObject -= damagePlayer;
+                        if (hpObject <= 0)
+                        {
+                            Console.WriteLine("You win!");
+                            world.CurrentLocation().ObjectOnLocation.ObjectIsAlive = false;
+                            break;
+                        }
+                        Console.WriteLine($"{objectName} did {damageObject} damage to you.");
+                        hpPlayer -= damageObject;
+                        if (hpPlayer <= 0)
+                        {
+                            Console.ForegroundColor = ConsoleColor.Red;
+                            Console.WriteLine("You're dead. Thanks for playing.");
```

Comment:
You forgot to restore console color before `Exit` again. Please call `ResetColor()` after the `WriteLine` but before `Environment.Exit`.